### PR TITLE
DrawableSpans ease-of-use

### PR DESCRIPTION
**Pull request by Hoikas**
This changeset is designed to ease working with `plDrawableSpans` objects. You get a bonus memory leak fix for free.

### CODE REVIEW

File: core/PRP/Geometry/plDrawableSpans.cpp [hunk 1]
```diff
@@ -35,6 +35,8 @@ plDrawableSpans::~plDrawableSpans()
         delete *group;
     for (auto span = fSpans.begin(); span != fSpans.end(); ++span)
         delete *span;
+    for (auto span = fSourceSpans.begin(); span != fSourceSpans.end(); ++span)
+        delete *span;
```
Comment (zrax) [inline core/PRP/Geometry/plDrawableSpans.cpp:39]:
```suggestion
    for (auto span : fSourceSpans)
        delete span
```

File: core/PRP/Geometry/plDrawableSpans.cpp [hunk 2]
```diff
@@ -907,3 +909,10 @@ size_t plDrawableSpans::addSourceSpan(plGeometrySpan* span)
     fSourceSpans.push_back(span);
     return fSourceSpans.size() - 1;
 }
+
+void plDrawableSpans::clearSourceSpans()
+{
+    for (auto span = fSourceSpans.begin(); span != fSourceSpans.end(); ++span)
+        delete *span;
```
Comment (zrax) [inline core/PRP/Geometry/plDrawableSpans.cpp:916]:
```suggestion
    for (auto span : fSourceSpans)
        delete span;
```

File: core/PRP/Geometry/plDrawableSpans.h [hunk 3]
```diff
@@ -190,6 +191,10 @@ class PLASMA_DLL plDrawableSpans : public plDrawable
     void clearTransforms();
     size_t addTransform(const hsMatrix44& l2w, const hsMatrix44& w2l,
                         const hsMatrix44& l2b, const hsMatrix44& b2l);
+    std::tuple<hsMatrix44, hsMatrix44, hsMatrix44, hsMatrix44> getTransform(size_t idx) const
+    {
+        return std::make_tuple(fLocalToWorlds[idx], fWorldToLocals[idx], fLocalToBones[idx], fBoneToLocals[idx]);
```
Comment (zrax) [inline core/PRP/Geometry/plDrawableSpans.h:196]:
```suggestion
        return std::make_tuple(fLocalToWorlds[idx], fWorldToLocals[idx],
                               fLocalToBones[idx], fBoneToLocals[idx]);
```

**Review by zrax — APPROVED** [review]

File: core/PRP/Geometry/plDrawableSpans.cpp (hunk 1 above)
Comment (Hoikas) [reply core/PRP/Geometry/plDrawableSpans.cpp:39]: Yeah, I was going for consistency with the rest of the file here :wink:

Comment (zrax) [reply core/PRP/Geometry/plDrawableSpans.cpp]: Hmm, fair point...  But I'm not opposed to modernizing new code even in files with older code ;)

Comment (zrax) [reply core/PRP/Geometry/plDrawableSpans.cpp]: And of course the build break is my fault (missing `;`)...  :/

**Pull request merged**